wordpress-mobile/WordPress-Android
# Feature/13066 ppn categories

Fixes #13066 

This is the main branch for the PPN select and Add Categories in the publish bottom sheet.

Home|Select|Add
---|---|---
<img width="480" src="https://user-images.githubusercontent.com/506707/97504498-5ee25700-194d-11eb-8202-deac5b172bcb.png">|<img width="480" src="https://user-images.githubusercontent.com/506707/97504488-57bb4900-194d-11eb-8780-36fdb86d6f36.png">|<img width="480" src="https://user-images.githubusercontent.com/506707/97504469-4ffba480-194d-11eb-91b2-7b330ad4bd0a.png">

**Notes**
- Unit tests are following in a separate task outside of this PR
- There is an existing API issue that will return the wrong term information for wordpress.com sites, thus sometimes checking the wrong category if adding a sub-category term name that is the same as another sub-cateogy term name. It is under investigation by the API team. 

Merge instructions
- Make sure #13256 has been merged
- Remove the Not Ready For Merge Status
- Merge as normal

**To test:**
The bulk of PRs reviewed as part of #13066 were tested, but it needs one more round of testing now that develop has been merged in.

- Launch the app
- Select Blog posts
- Select Edit on a post
- From the edit post view, tap the UPDATE action to launch the bottom sheet
- Categories are shown on the Categories row
- If the post has categories assigned, you will see them listed on the home view. If not, you will see Not Set or Uncategorized
- Select Categories is shown when user taps on Categories
- 
- Tap the categories row to launch the Select categories view
- Select/unselect a few categories - but note which ones
- Tap the Add New Category button "+" on the title bar
- Enter a new category and tap "Add Category Button" Note: A new category is NOT saved on back press - the explicit tap on the action button is needed)
- Back on the select category view, the category you have added is visible and checked
- Take note of the toast which will tell you if the action succeeded or failed.
- The categories you unselected/selected will be shown properly
- Tap the <- Back arrow or the device back button to save your options
- The categories row on the home screen should reflect the selections you made


PR submission checklist:
